Readline implementation

LithiumOx opened the pull request.

LithiumOx committed:
🔧 chore(.gitmodules): add submodule for readline
📦 feat(readline): add readline submodule with commit 7274faa

LithiumOx committed:
🐛 fix(.gitignore): add readline/ directory to the tracked files
🐛 fix(Makefile): fix typo in INCLUDES variable
✨ feat(Makefile): add support for building and linking with readline library

🔧 fix(minishell.h): update last modified timestamp in header file
🔧 fix(main.c): update last modified timestamp in main file
✨ feat(main.c): add basic interactive shell functionality

LithiumOx committed:
🔧 chore(Makefile): remove unnecessary line in Makefile
🔄 chore(libft): update libft submodule to commit b40a7c7

LithiumOx committed:
🔄 chore(libft): update libft submodule to commit 9e1d039

JdeBaaij commented:
Bij het runnen van Make
"make[1]: *** No rule to make target `static'.  Stop.
make: *** [readline/libreadline.a] Error 2"

LithiumOx commented:
> Bij het runnen van Make "make[1]: *** No rule to make target `static'. Stop. make: *** [readline/libreadline.a] Error 2"

Je hebt helemaal gelijk maar ik heb moeite met ./configure called in de folder zonder de compile te laten plaatsvinden in de main project directory. Hierdoor komen er allemaal files die er eigenlijk niet in horen te komen.

LithiumOx committed:
🔧 chore(Makefile): improve build process for Readline library
🔥 chore(Makefile): remove unnecessary files and directories during cleanup
🔧 chore(Makefile): remove $(LIBS) file during fclean
🔧 chore(main.c): update timestamp in file header

LithiumOx commented:
@JdeBaaij Zou je het eventueel met de laatste commit kunnen proberen?

JdeBaaij commented:
> @JdeBaaij Zou je het eventueel met de laatste commit kunnen proberen?

🌈  Building Readline...        /bin/sh: ./configure: No such file or directory
make: *** [readline/libreadline.a] Error 127

Is het een probleem om de compile te laten plaatsvinden in de main project directory?

mijn submodule readline is leeg, ik denk dat ik misschien een fout heb gemaakt met het git clonen, gonna check

JdeBaaij reviewed (approved):
Oke dus na wat gekloot met GIT werkt het. Make / make fclean / make werkt.
./minishell opent een continuous readline, nice!
